pkg/logger/logger.go

```diff
@@ -176,7 +193,8 @@ func LogMemoryStats(ctx context.Context) {
 		memUsage.GarbageCollections,
 	)
 
-	color.Cyan(statsMessage)
+	log := l.log.With(append(NetworkIdentifierFields(l.network), zap.String("method", "LogMemoryStats"))...)
```

Comment:
This creates a childLogger - every it is called which is a bad design. we should called this only once and then use it for logging forever which the fields won't change for a particular run.